src/dndbeyond/content-scripts/character.js

```diff
@@ -873,24 +873,79 @@ function displayAction(paneClass) {
     });
 }
 
+function handleCustomText(paneClass) {
+    // Function Objective: Search for and handle block of VTT Message text...
+    //...in pane Notes or Description
+    const outCustomRollFuncs = {
+        before: [],
+        replace:[],
+        after:  [],
+        roll: (customTextFuncList, normalRollFunction=undefined) => {
+            if (customTextFuncList.length > 0) {
+                for (let i = 0; i < customTextFuncList.length; i++) {
+                    customTextFuncList[i]();
+                }
+            } else if (normalRollFunction) {
+                normalRollFunction();
+            }
+        },
+    };
+    const regexp = /```\[([^\]]*)\]\s*((\S|\s)*?)\s*```/g;
+    const rollOrderTypes = ["before", "after", "replace"];//(relative to normal roll msg)
+    const paneText = descriptionToString($(`.${paneClass}`).find(".ddbc-property-list__property:contains('Notes:'), .ct-action-detail__description, .ct-spell-detail__description, .ct-item-detail__description, .ddbc-action-detail__description, .ddbc-spell-detail__description, .ddbc-item-detail__description"));
+    console.log("Pane Text: " + paneText);
+    const matchedText = [...paneText.matchAll(regexp)];
+    //Handle Matched Text
+    if (matchedText) {
+        console.log("Matched Text:\n"+matchedText);
+        for (let i = 0; i < matchedText.length; i++) {
+            const cMatchType = matchedText[i][1].toLowerCase();
+            const cMatchText = matchedText[i][2];
+            // Determine if we should prevent the default action/message from being sent to the VTT
+            console.log("Note Match #"+i+" (Provided Type: "+cMatchType+"):\n"+cMatchText);
+            let rollOrder = rollOrderTypes[0];
+            if (rollOrderTypes.includes(cMatchType)) {
+                rollOrder = cMatchType;
+            } else {
+                console.warn(`Invalid CustomText roll order specified ("${cMatchType}"). Defaulting to "${rollOrder}" instead.`)
+            }
```

Comment:
I don't like that you match all `[anything]` content and default to `before`. If it's invalid, then it should be ignored.